src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -0,0 +1,256 @@
+    B1, B2, B3 = magnetic_field(u, equations)
+    v1_plus, v2_plus, v3_plus, vk1_plus, vk2_plus, vk3_plus = charge_averaged_velocities(u,
+                                                                                         equations)
+
+    s = zero(MVector{nvariables(equations), eltype(u)})
+
+    for k in eachcomponent(equations)
+        rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, u, equations)
+        v1 = rho_v1 / rho
+        v2 = rho_v2 / rho
+        v3 = rho_v3 / rho
+        v1_diff = v1_plus - v1
+        v2_diff = v2_plus - v2
+        v3_diff = v3_plus - v3
+        r_rho = charge_to_mass[k] * rho
+        s2 = r_rho * (v2_diff * B3 - v3_diff * B2)
+        s3 = r_rho * (v3_diff * B1 - v1_diff * B3)
+        s4 = r_rho * (v1_diff * B2 - v2_diff * B1)
+        s5 = v1 * s2 + v2 * s3 + v3 * s4
+
+        set_component!(s, k, 0, s2, s3, s4, s5, equations)
+    end
+
+    return SVector(s)
+end
+
+"""
+    electron_pressure_zero(u, equations::AbstractIdealGlmMhdMultiIonEquations)
+
+Returns the value of zero for the electron pressure. Needed for consistency with the 
+single-fluid MHD equations in the limit of one ion species.
+"""
+function electron_pressure_zero(u, equations::AbstractIdealGlmMhdMultiIonEquations)
+    return zero(u[1])
+end
+
+"""
+    v1, v2, v3, vk1, vk2, vk3 = charge_averaged_velocities(u,
+                                                       equations::AbstractIdealGlmMhdMultiIonEquations)
+
+
+Compute the charge-averaged velocities (`v1`, `v2`, and `v3`) and each ion species' contribution
+to the charge-averaged velocities (`vk1`, `vk2`, and `vk3`). The output variables `vk1`, `vk2`, and `vk3`
+are `SVectors` of size `ncomponents(equations)`.
+"""
+@inline function charge_averaged_velocities(u,
+                                            equations::AbstractIdealGlmMhdMultiIonEquations)
+    total_electron_charge = zero(real(equations))
+
+    vk1_plus = zero(MVector{ncomponents(equations), eltype(u)})
+    vk2_plus = zero(MVector{ncomponents(equations), eltype(u)})
+    vk3_plus = zero(MVector{ncomponents(equations), eltype(u)})
+
+    for k in eachcomponent(equations)
+        rho, rho_v1, rho_v2, rho_v3, _ = get_component(k, u, equations)
+
+        total_electron_charge += rho * equations.charge_to_mass[k]
+        vk1_plus[k] = rho_v1 * equations.charge_to_mass[k]
+        vk2_plus[k] = rho_v2 * equations.charge_to_mass[k]
+        vk3_plus[k] = rho_v3 * equations.charge_to_mass[k]
+    end
+    vk1_plus ./= total_electron_charge
+    vk2_plus ./= total_electron_charge
+    vk3_plus ./= total_electron_charge
+    v1_plus = sum(vk1_plus)
+    v2_plus = sum(vk2_plus)
+    v3_plus = sum(vk3_plus)
+
+    return v1_plus, v2_plus, v3_plus, SVector(vk1_plus), SVector(vk2_plus),
+           SVector(vk3_plus)
+end
+
+"""
+    get_component(k, u, equations::AbstractIdealGlmMhdMultiIonEquations)
+
+Get the hydrodynamic variables of component (ion species) `k`.
+"""
```

Comment:
Not sure if these should be documented/exported.

Reply:
Following the discussion [here](https://github.com/trixi-framework/Trixi.jl/pull/2196#discussion_r1879714530), this function is not exported. However, I prefer to keep the docstring as it can improve the user experience when dealing with the multi-ion GLM-MHD equations. Does anything speak against having a docstring here? @sloede

Reply:
In the past, we have considered everything with a docstring (exported or not) to be the "official" API of Trixi.jl. That is, if you change such a function, you also need to create a breaking release.

Thus, to be on the safe side, add an "experimental function" disclaimer for now (there are plenty around, just copy&paste one of them) such that you can change the function later at will 😊

Side note: With Julia v1.11 and the `public` keyword, this convention will eventually become irrelevant, since we'll be able to explicitly express what's part of the API and what isn't.

Reply:
Added experimental admonitions in b8e4a290393512c85b024f7d956c53576c149f1d.